### DISCUSSION TITLE
Image Enhancer with combined Upscale feature

### DISCUSSION THREAD

**Discussion by mel870** [lllyasviel/Fooocus#1068] [Ideas]
Recently, I have seen some web apps have incorporated features to enhance AI image quality.  Krea and Magnific. It appears they may be utilizing Stable Diffusion on the backend, supplemented with a maybe a  custom highres fix script and an upscale model.

I believe such a feature could significantly enrich Fooocus. To further empower advanced users, there could be options to customize in the developer settings. This would allow for a balance between simplicity and customization, catering to both novice and expert users.

[Krea Upscale & Enhance](https://twitter.com/krea_ai/status/1729785367552004438)
[Magnific Upscale & Enhance](https://magnific.ai/ )

**Comment by jclevet** [comment]
Here you can discover the best upscaler models by type of images that can be used on ChaiNNer : 
https://phhofm.github.io/upscale/favorites.html
SwinIR-L + CodeFormer is magic for photos with faces.

**Reply by mel870** [reply]
I mainly do art styles and not really people. Also, it's fairly easy to fix faces though. It seems like the services I shared enhance the images including removing artifacts versus just upscaling. Magnific has examples of a variety of images on twitter and not just people. I'm assuming it's a tweaked hi-res fix. I sometimes take Fooocus images into Auto1111 and use hi-res there and recently Kohya's. Also,  I've noticed the new TurboVisionXL model which uses SD Turbo has fewer artifacts when testing,  so fairly soon something like hi-res may not be needed, but it would be great to have an enhance image workflow in Fooocus.
